inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -703,11 +702,71 @@ Parameter clDNNEngine::GetMetric(const std::string& name, const std::map<std::st
     } else if (name == METRIC_KEY(RANGE_FOR_STREAMS)) {
         std::tuple<unsigned int, unsigned int> range = std::make_tuple(1, 2);
         IE_SET_METRIC_RETURN(RANGE_FOR_STREAMS, range);
+    } else if (name == GPU_METRIC_KEY(MAX_BATCH_SIZE)) {
+        const auto& config = _impl->m_configs.GetConfig(device_id);
+        auto n_streams = config.throughput_streams;
+        auto available_device_mem = device_info.max_global_mem_size;
+        int64_t max_batch_size = 0;
+
+        if (options.find("CNN_NETWORK") == options.end()) {
```

Comment:
Do we really need to use CNNNetwork? I'd prefer something like `"MODEL_PTR" -> ngraph::Function*`

Reply:
Fixed to use ngraph Function